newrelic/resource_newrelic_synthetics_private_location.go

```diff
@@ -0,0 +1,193 @@
+			},
+		},
+	}
+}
+
+func resourceNewRelicSyntheticsPrivateLocationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+	var diags diag.Diagnostics
+
+	description := d.Get("description").(string)
+	name := d.Get("name").(string)
+	verifiedScriptExecution := d.Get("verified_script_execution").(bool)
+	res, err := client.Synthetics.SyntheticsCreatePrivateLocationWithContext(ctx, accountID, description, name, verifiedScriptExecution)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(res.Errors) > 0 {
+		for _, err := range res.Errors {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  err.Description,
+			})
+		}
+	}
+
+	if len(diags) > 0 {
+		return diags
+	}
+	d.SetId(string(res.GUID))
+	_ = d.Set("domain_id", res.DomainId)
+	_ = d.Set("key", res.Key)
+	_ = d.Set("location_id", res.LocationId)
+	_ = d.Set("guid", res.GUID)
+
+	return nil
+}
+
+func resourceNewRelicSyntheticsPrivateLocationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfig).NewClient
+	log.Printf("[INFO] Reading New Relic Synthetics Private Location %s", d.Id())
+
+	guid := common.EntityGUID(d.Id())
+	resp, err := client.Entities.GetEntity(guid)
+	if err != nil {
+		if _, ok := err.(*errors.NotFound); ok {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	setCommonSyntheticsPrivateLocationAttributes(resp, d)
+	return nil
+}
+
+func setCommonSyntheticsPrivateLocationAttributes(v *entities.EntityInterface, d *schema.ResourceData) {
+	switch e := (*v).(type) {
+	case *entities.GenericEntityOutline:
+		_ = d.Set("guid", e.GUID)
+		_ = d.Set("name", e.Name)
+
+	}
+}
+
+func resourceNewRelicSyntheticsPrivateLocationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	var diags diag.Diagnostics
+	description := d.Get("description").(string)
+	guid := synthetics.EntityGUID(d.Id())
+	verifiedScriptExecution := d.Get("verified_script_execution").(bool)
+	res, err := client.Synthetics.SyntheticsUpdatePrivateLocation(description, guid, verifiedScriptExecution)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(res.Errors) > 0 {
+		for _, err := range res.Errors {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  err.Description,
+			})
+		}
+	}
+
+	if len(diags) > 0 {
+		return diags
+	}
+
+	return nil
+}
+
+func resourceNewRelicSyntheticsPrivateLocationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+
+	client := providerConfig.NewClient
+	var diags diag.Diagnostics
+	guid := synthetics.EntityGUID(d.Id())
```

Comment:
Most other resources use `common.EntityGUID`, but not sure if that matters. cc @mbazhlekova @sanderblue

Reply:
Technically it doesn't matter since they're both typed as strings, but we probably should stick with the `common.EntityGUID`. Good call on that, I forgot we had made that common package for this sort of thing.